src/main/java/application/App.java

```diff
@@ -0,0 +1,42 @@
+package application;
+import http_server.RequestParser;
+import http_server.Response;
+import http_server.Server;
+import http_server.Router;
+
+public class App {
+    static Router app;
+    static Server server;
+
+    public static void main(String args[]) {
+        app = new Router();
+        createRoutes();
```

Comment:
So, here, you're creating an instance variable, then calling a method, which mutates it in a different part of the file.  I try to suggest avoiding setup patterns like this, which can be tough in Java, because the language and the learning resources for the language both strongly encourage it.

Off the top of my head, here's some of the the issues I can see happening down the road with this kind of setup:

* If another dev tries to move `createRoutes` somewhere, they have to know it interacts with `app` (and they have to know that `createRoutes` changes it).
* If another dev wants to know what's going on with this code, they have to go back and forth between two spots to understand it.

In these scenarios, "another dev" is usually just future you :)

Honestly those are both pretty microscopic _here_, so I wouldn't be too worried about it in this specific spot, but I think it's easy enough to avoid issues like that by just keeping collaborating bits of code close to each other.  That way you just sort of reduce the thought required to grok what's going on.

```java
// In .main:
app = createRouter();

// In .createRoutes, renamed to .createRouter:
private static Router createRouter() {
  Router app = new Router();
  // ... setup as normal
  return app;
}
```

Also, there's no shame in just having a long method every now and again, especially if the length is mostly due to some setup work.  It happens, and often times the best refactors emerge when you keep collaborating code co-located (even if you find it unpleasant).